astropy/modeling/functional_models.py

```diff
@@ -1370,3 +1454,109 @@ def custom_model_1d(func, func_fit_deriv=None):
             "All parameters must be keyword arguments")
 
     return custom_model(func, fit_deriv=func_fit_deriv)
+
+
+class Sersic2D(Fittable2DModel):
+    r"""
+    Two dimensional Sersic surface brightness profile.
+
+    Parameters
+    ----------
+    amplitude : float
+        Central surface brightness, within r_eff. 
+    r_eff : float
+        Effective (half-light) radius
+    n : float
+        Sersic Index.
+    x_0 : float, optional
+        x position of the center.
+    y_0 : float, optional
+        y position of the center. 
+    ellip : float, optional
+        Ellipticity.
+    theta : float, optional
+        Rotation angle in radians, counterclockwise from
+        the positive x-axis. 
+
+    See Also
+    --------
+    Gaussian2D, Moffat2D, Lorentz2D
+
+    Notes
+    -----
+    Model formula:
+
+    .. math::
+
+        I(x,y) = I(r) = I_e\exp\left[-b_n\left(\frac{r}{r_{e}}\right)^{(1/n)}-1\right]
+
+    The constant :math:`b_n` is defined such that :math:`r_e` contains half the total 
+    luminosity, and can be solved for numerically.  
+
+    .. math:: 
+
+        \Gamma(2n) = 2\gamma (b_n,2n)
+
+    Examples
+    --------
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        from astropy.modeling.models import Sersic2D
+        import matplotlib.pyplot as plt
+        from astropy.visualization import LogStretch
+        from astropy.visualization.mpl_normalize import ImageNormalize
+
+        x,y = np.meshgrid(np.arange(100),np.arange(100))
+
+        mod = Sersic2D(amplitude = 1, r_eff = 25, n=4, x_0=50, y_0=50, ellip=.5,theta=-1)
+        img = mod(x,y)
+        norm = ImageNormalize(vmin=img.min(), vmax=img.max(), stretch=LogStretch())
+
+        plt.figure()
+        plt.imshow(img, origin='lower', interpolation='nearest', cmap='binary_r', norm=norm)
+        plt.xlabel('x', fontsize=16)
+        plt.ylabel('y', fontsize=16)
+        cbar=plt.colorbar()
+        cbar.set_label('log I(x,y)', rotation=270, labelpad=25, fontsize=16)
+        plt.show()
+
+    References
+    ----------
+    .. [1] http://ned.ipac.caltech.edu/level5/March05/Graham/Graham2.html
+    """
+
+    amplitude = Parameter(default=1)
+    r_eff = Parameter(default=1)
+    n = Parameter(default=4)
+    x_0 = Parameter(default=0)
+    y_0 = Parameter(default=0)
+    ellip = Parameter(default=0)
+    theta = Parameter(default=0)
+    _gammaincinv = None
+
+    def __init__(self, amplitude=amplitude.default, r_eff=r_eff.default,
+                 n=n.default, x_0=x_0.default, y_0=y_0.default, ellip=ellip.default,
+                 theta=theta.default, **kwargs):
+        try:
+            from scipy.special import gammaincinv
+            self.__class__._gammaincinv = gammaincinv
+        except ValueError:
+            raise ImportError("Sersic2D model requires scipy > 0.11.")
```

Comment:
I think this is probably inspired by how the `AiryDisk2D` model does something similar to get a zero of a Bessel function IIRC.  That reminds me that I'm not super happy with it--I might like to improve this in the future, but it's totally fine for now.  This is more a reminder to myself.